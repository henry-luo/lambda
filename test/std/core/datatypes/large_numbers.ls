// Test: Large Number Operations
// Category: core/datatypes
// Type: positive
// Expected: 1000000

// Test operations with large numbers
let a = 1000
a
let b = 1000
b
a * b
