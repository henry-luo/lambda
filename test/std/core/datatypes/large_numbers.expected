09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/large_numbers.ls...
[script:list]
  [content:list[3]]
    no declare
    item:
    [let stam:null]
      declare:
      [assign expr:a:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [let stam:null]
      declare:
      [assign expr:b:int]
        [primary expr:int,const:1]
          [const: int]
    item:
    [binary expr *.5:int]
      [primary expr:int,const:1]
        [ident:a:int,const:1]
      [primary expr:int,const:1]
        [ident:b:int,const:1]
09:15:42 [NOTICE] compiling C code in 'test/std/core/datatypes/large_numbers.ls' to MIR
C2MIR init end           -- 70 usec
test/std/core/datatypes/large_numbers.ls:1:2: warning -- unknown pragma
    preprocessor tokens -- 2426, parse tokens -- 1458
  C2MIR preprocessor end    -- 541 usec
  C2MIR parser end          -- 658 usec
  C2MIR context checker end -- 738 usec
  C2MIR generator end       -- 797 usec
C2MIR compiler end                -- 834 usec
09:15:42 [NOTICE] Generating native code...
##### Script 'test/std/core/datatypes/large_numbers.ls' executed: #####
1000000
